src/Matchers/UrlMatcher.php

```diff
@@ -7,7 +7,7 @@
 
 class UrlMatcher implements FakeResponseMatcherContract
 {
-    public function __construct(private string $url, private ?string $method = 'GET')
+    public function __construct(private string $url, private ?string $method = null, private ?string $body = null)
```

Comment:
It might make sense to do `?array|string` here for convenience? 

If an array is passed in, maybe we json encode it for the user?

Reply:
Also, i think defaulting to 'GET' here is fine still

Reply:
We are matching with ->getBody()->getContents() which is always a string so should make sure body is also a string 

/**
     * Returns the remaining contents in a string
     *
     * @return string
     * @throws \RuntimeException if unable to read or an error occurs while
     *     reading.
     */
    public function getContents();

Reply:
We can't default to GET as in the legacy tests they don't match method and are currently working for GET and POST requests. Penny noticed this bit was breaking existing tests so I made it optional to be more easily ported